### PR TITLE
search rum action

### PR DESCRIPTION
**Pull request by bgdeutsch**
### What does this PR do?
Sends a custom RUM action based on user searches

### Motivation
More data on how users are searching is useful

### Preview 
https://docs-staging.datadoghq.com/brian.deutsch/search-rum-action

### Testing
- search from the homepage by either clicking `Submit` or pressing enter.
- search from the side nav on any page including clicking through one of the links in the dropdown
- click through one of the links from the search page
- custom rum actions should be sent to Datadog.   Try using the RUM actions tab in the explorer here: https://dd-corpsite.datadoghq.com/rum/explorer?query=%40application.id%3A3493b4e7-ab12-4852-8836-ba96af7bc745%20env%3Apreview%20%40session.type%3Auser%20%40action.type%3Acustom%20%40type%3Aaction&cols=&viz=stream&from_ts=1682261659467&to_ts=1682348059467&live=true to see the custom action being sent.

- there should be no console errors present and other UI javascript functionality is unaffected.  Please make sure to QA some JS elements (language/site dropdowns, mobile menu, TOC, tabs, etc)
- search should continue work including from the JA and FR sites and RUM actions should be posted from the translated sites as well.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: assets/scripts/components/algolia.js [hunk 1]
```diff
@@ -155,17 +165,29 @@ function loadInstantSearch(asyncLoad) {
...
+
                 do {
-                    if (target === searchBoxContainerContainer) {
-                        return;
-                    }
+                    if (target === searchBoxContainerContainer) return;
+
                     target = target.parentNode;
+
+                    if (target && target.href) {
```
[the remark this answers is not in the document]
Comment (bgdeutsch) [reply assets/scripts/components/algolia.js]: thank you @DavidWeid!   made a change here based on your feedback if you could have another look.

**Review by maycmlee — APPROVED** [review]
I went through the testing steps, testing search for EN/JA/FR, spot checked tabs and site dropdown menu, and it looks good to me! Is this good to merge?

File: assets/scripts/components/algolia.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DavidWeid) [reply assets/scripts/components/algolia.js]: LGTM

File: assets/scripts/components/algolia.js [hunk 2]
```diff
@@ -155,16 +165,28 @@ function loadInstantSearch(asyncLoad) {
 
             const handleSearchbarSubmitClick = () => {
                 if (aisSearchBoxInput.value) {
+                    sendSearchRumAction(search.helper.state.query);
                     window.location.pathname = searchPathname;
```
Comment (StefonSimmons) [inline assets/scripts/components/algolia.js:169]: does sending the query refresh the page? is that why we need to update the window pathname?

**Pull request merged**